src/main/java/com/example/demo/orderItem/OrderItem.java

```diff
@@ -0,0 +1,47 @@
+package com.example.demo.orderItem;
+
+import com.example.demo.order.Order;
+import com.example.demo.shop.Product;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import javax.persistence.*;
+import java.time.LocalDateTime;
+
+@Entity
+@Table(name = "order_item")
+@Getter
+@Setter
+@NoArgsConstructor(access = AccessLevel.PROTECTED)  // 생성자를 다른 계층에서 함부로 생성하지 못하도록 막아준다.
+public class OrderItem {
+
+    @Id
+    @GeneratedValue
+    @Column(name = "order_item_id")
+    private Long id;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "product_id")
+    private Product product;
+
+    @JsonIgnore
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "order_id")
+    private Order order;
+
+    private int orderPrice;     // 주문 가격
+
+    // 생성 메서드
+    // 먼저 product 랑 orderItem 이랑 묶어준다.
+    public static OrderItem createOrderItem(Product product, int price){
+        OrderItem orderItem = new OrderItem();
```

Comment:
여기 int price로 처리안해도 될것 같은데, Product , Product.getPrice해서 쓰는거면 Product  받아서 걍 setOrederPrice 넣는게 나은 것 같아보입니다.!

```suggestion
        OrderItem orderItem = OrderItem.createOrderItem(product,product.getPrice());
```
요기에서 보면 getPrice해서 그 값 억지로 넣어서 처리하기보다는 내부적으로 처리하는 게 좋을 것 같아요

Reply:
파라미터로 product만 받아서 여기서 getPrice로 가격 설정하자는 얘기지?

Reply:
넹 맞습니당

Reply:
반영함쓰~